test/test.options.ts

```diff
@@ -99,17 +99,19 @@ describe('Options', () => {
   });
 
   it('should use the same per-API setting twice', async () => {
```

Comment:
Can you think of any edge cases we might be missing a test for, e.g., we're happy with our tests for per-request options.

Reply:
I think we're in pretty ok shape here.  You can always have more tests, but we test a pretty broad range.